### DISCUSSION TITLE
std::ios::hex doesn't work for SPARTA_ADDPAIR

### DISCUSSION THREAD

**Discussion by quasiquote1** [sparcians/map#274] [Q&A]
Hello,  
I created my own PairDefinition to be used by the Helios pipeline viewer, and I followed the same example as in ExampleInst.hpp (ExampleInstPairDef).  

I am trying to display my addresses in hex instead of decimal, but even after using std::ios::hex I am unable to do this - it always comes out as decimal in the Helios pipeviewer tool.

Is this a known bug?  Has this been fixed?   
Thanks!

**Comment by klingaard** [comment]
Hmmm...  it works in our models...  I'll take a look and see what might be the problem.

**Comment by klingaard** [comment]
After chatting with a few folks that are more familiar with Argos, I found out that there was a hack added to Argos to explicitly convert named fields to hex for display.  

https://github.com/sparcians/map/blob/7be1f32a3e6df3dc81c414221b8e970525c39540/helios/pipeViewer/transactiondb/src/Reader.hpp#L566-L575

Naturally, this isn't the preferred way -- we should be using the preset IOS format fields in the database.  I'll file a bug and see if we can add this functionality.  We could use it too.

**Comment by klingaard** [comment]
https://github.com/sparcians/map/issues/277
